src/cpp/middleware/ced/CedEntities.cpp

```diff
@@ -0,0 +1,229 @@
+// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <uxr/agent/middleware/ced/CedEntities.hpp>
+#include <chrono>
+
+namespace eprosima {
+namespace uxr {
+
+/**********************************************************************************************************************
+ * CedTopicManager
+ **********************************************************************************************************************/
+std::unordered_map<int16_t, std::unordered_map<std::string, std::shared_ptr<CedTopicImpl>>> CedTopicManager::topics_;
+std::mutex CedTopicManager::mtx_;
+
+bool CedTopicManager::register_topic(
+        const std::string& topic_name,
+        int16_t domain_id,
+        std::shared_ptr<CedTopicImpl>& topic)
+{
+    std::unique_lock<std::mutex> lock(mtx_);
+    auto it = topics_[domain_id].find(topic_name);
+    if (topics_[domain_id].end() == it)
+    {
+        topic = std::shared_ptr<CedTopicImpl>(new CedTopicImpl(topic_name, domain_id));
```

Comment:
std::make_shared?

Reply:
Can we use `std::unordered_map::emplace`?

Reply:
Done!